js/DataService.js

```diff
@@ -232,41 +215,31 @@ angular.module('PathOfDamage')
       scope.sections.mitigation.resistance.chaos = parseIntOrNull(sections[13]);
       scope.sections.mitigation.healthPool = parseIntOrNull(sections[14]);
     },
-    decodeTable: function (tableString, damageTypes, extraParsing) {
+    decodeTable: function (table, tableString, damageTypes) {
       if (tableString) {
-        var table = [];
         var rows = tableString.split(ROW_DELIMITER);
         for (var i = 0; i < rows.length; i++) {
           var values = rows[i].split(VALUE_DELIMITER);
-          var tableEntry = {
-            enabled: values[0].slice(0, 1) === '1',
-            name: values[0].slice(1),
-            value: parseIntOrNull(values[1])
-          };
-          if (extraParsing) {
-            extraParsing(values, tableEntry, damageTypes);
-          }
-          table.push(tableEntry);
+          table.addRow(
+              values[0].slice(0, 1) === '1',
+              values[0].slice(1),
+              parseIntOrNull(values[1]),
+              this.decodeElements(values[2], damageTypes)
+          );
         }
-        return table;
-      }
-    },
-    decodeElement: function (values, tableEntry, damageTypes) {
-      if (values[2]) {
-        tableEntry.element = damageTypes[values[2]];
       }
     },
-    decodeTypes: function (values, tableEntry, damageTypes) {
-      if (values[2]) {
-        var types = [];
-        var booleans = parseInt(values[2], 36).toString(2).split('');
+    decodeElements: function (encodedElements, elementTypes) {
```

Comment:
App.js doesn't use this, we should just have this in class scope instead of making it part of the returned object. Also some of the other en/decoders.